javaagent-instrumentation-api/build.gradle.kts

```diff
@@ -7,10 +7,8 @@ plugins {
 group = "io.opentelemetry.javaagent"
 
 dependencies {
-  api("io.opentelemetry:opentelemetry-api")
-  compileOnly("io.opentelemetry:opentelemetry-sdk")
+  api(project(":instrumentation-api"))
```

Comment:
`javaagent-instrumentation-api` directly extends `instrumentation-api` with a few javaagent-specific usecases, so it makes sense to expose `instrumentation-api` as API dependency here.